src/test/java/de/zalando/aruha/nakadi/controller/EventStreamControllerTest.java

```diff
@@ -277,6 +287,57 @@ public void whenExceptionIsThrownThenInternalServerError() throws NakadiExceptio
         assertThat(responseToString(responseBody), jsonHelper.matchesObject(expectedProblem));
     }
 
+    @Test
+    public void reportCurrentNumberOfConsumers() throws Exception {
+        when(topicRepositoryMock.topicExists(anyString())).thenReturn(true);
+        final EventStream eventStream = mock(EventStream.class);
+
+        // block to simulate the streaming until thread is interrupted
+        Mockito.doAnswer(invocation -> {
+            while (!Thread.interrupted()) { Thread.sleep(100); }
+            return null;
+        }).when(eventStream).streamEvents();
+        when(eventStreamFactoryMock.createEventStream(any(), any(), any())).thenReturn(eventStream);
+
+        // "connect" to the server
+        final StreamingResponseBody responseBody = controller.streamEvents(TEST_EVENT_TYPE, 0, 0, 0, 0, 0, null,
+                requestMock, responseMock);
+
+
+        final LinkedList<Thread> clients = new LinkedList<>();
+        final Counter counter = metricRegistry.counter(metricNameFor(TEST_EVENT_TYPE, EventStreamController.CONSUMERS_COUNT_METRIC_NAME));
+
+        // create clients...
+        for (int i=0; i<3; i++) {
```

Comment:
I have nothings against the number 3, but I feel that only one client would be enough to test it. I'm not suggesting that you should change, but just would like to share my thoughts on this subject.